TeapotTests/TeapotTests.swift

```diff
@@ -165,4 +165,21 @@ class TeapotTests: XCTestCase {
 
         self.waitForExpectations(timeout: 20.0)
     }
+
+    func testCancelRequest() {
+        let expectation = self.expectation(description: "Get")
+
+        let task = self.teapot?.get("/get") { (result: NetworkResult) in
+            XCTFail() // we're cancelling the task, it should never complete.
+        }
+
+        task?.cancel()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 15.0) {
+            // we wait 15s, to be sure that the request is never completed.
```

Comment:
wouldn't it make more sense to check the [URLSessionTask.state](https://developer.apple.com/documentation/foundation/urlsessiontask.state) is either `canceling` or `completed` after calling `cancel` instead of waiting for 15 extra seconds?

Reply:
🤦🏽‍♂️ was doing the 15s for myself. Forgot to change back before pushing, hehe.

Reply:
Much nicer now!